config/src/main/java/com/quorum/tessera/config/constraints/ServerConfigsValidator.java

```diff
@@ -1,45 +1,57 @@
 package com.quorum.tessera.config.constraints;
 
 import com.quorum.tessera.config.AppType;
+import com.quorum.tessera.config.Config;
 import com.quorum.tessera.config.ServerConfig;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import javax.validation.ConstraintValidator;
 import javax.validation.ConstraintValidatorContext;
-import java.util.List;
 import java.util.Map;
 import java.util.stream.Collectors;
 
-public class ServerConfigsValidator implements ConstraintValidator<ValidServerConfigs, List<ServerConfig>> {
+public class ServerConfigsValidator implements ConstraintValidator<ValidServerConfigs, Config> {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(ServerConfigsValidator.class);
 
     @Override
-    public boolean isValid(List<ServerConfig> serverConfigs, ConstraintValidatorContext constraintContext) {
-        if(serverConfigs == null) {
+    public boolean isValid(Config config, ConstraintValidatorContext constraintContext) {
+        if (config == null || config.getServerConfigs() == null || config.getServerConfigs().isEmpty()) {
```

Comment:
I don't think this should return true in the last case (`config.getServerConfigs().isEmpty()`). This allows the application to start up when no servers are defined, when we mandate that at least one P2P server should be there. 

The reason it is okay to have `config.getServerConfigs() == null` is because that is checked by the `@ValidEitherServerConfigsOrServer` annotation. Maybe we should add a comment here saying that since it is rather indirect.